.github/workflows/ci-for-diff-ns.yaml

```
@@ -49,24 +49,49 @@ jobs:
           sleep 5
           kubectl rook_ceph -o test-operator -n test-cluster rbd ls replicapool
 
-          # for testing osd purge scale the osd deplyment
-          kubectl --namespace test-cluster scale deploy/rook-ceph-osd-0 --replicas=0
-          # we need to sleep so the osd will be marked down before purging the osd
-          sleep 5
-          kubectl-rook_ceph -o test-operator -n test-cluster rook purge-osd 0 --force
-          kubectl rook_ceph -o test-operator -n test-cluster health
-
           # for testing start-debug and stop-debug deployment
           # mon
           kubectl rook_ceph -o test-operator -n test-cluster debug start rook-ceph-mon-a
-          # sleep till the debug pod get running
-          sleep 5
+          # check if debug mon is started
+          mons=$(kubectl get pod -l app=rook-ceph-mon -n test-cluster -o jsonpath='{.items[*].metadata.name}'| grep debug)
+          for val in $mons; do
+            if [[ $val == *"debug"* ]];
+            then
+              debug_mon=$val
+            fi  
+          done
+          timeout 60 sh -c 'until [ "$(kubectl get pod $debug_mon -n test-cluster -o jsonpath='{.status.phase}'| grep -c "Running")" -eq 1 ]; do echo "waiting for pod rook-ceph-mon-a-debug to get created " && sleep 1; done'
           kubectl rook_ceph -o test-operator -n test-cluster debug stop rook-ceph-mon-a
           # osd
           kubectl rook_ceph -o test-operator -n test-cluster debug start rook-ceph-osd-0
-          # sleep till the debug pod get running
-          sleep 5
+          # check if debug osd is started
+          osds=$(kubectl get pod -l app=rook-ceph-osd -n test-cluster -o jsonpath='{.items[*].metadata.name}'| grep debug)
+          for val in $osds; do
+            if [[ $val == *"debug"* ]];
+            then
+              debug_osd=$val
+            fi  
+          done
+          timeout 60 sh -c 'until [ "$(kubectl get pod $debug_osd -n test-cluster -o jsonpath='{.status.phase}'| grep -c "Running")" -eq 1 ]; do echo "waiting for pod rook-ceph-osd-0-debug to get created " && sleep 1; done'
           kubectl rook_ceph -o test-operator -n test-cluster debug stop rook-ceph-osd-0
+          
+          # check if osd 0 is started
+          osds=$(kubectl get pod -l app=rook-ceph-osd -n test-cluster -o jsonpath='{.items[*].metadata.name}'| grep 0)
```

There is a label for the osd id you can query, but you better confirm the label name, i'm not sure if this is correct
```suggestion
          osds=$(kubectl get pod -l app=rook-ceph-osd,osd-id=0 -n test-cluster -o jsonpath='{.items[*].metadata.name}'| grep 0)
```

---

Okay, thanks